src/lib/time-picker/time-picker-foundation.ts

```diff
@@ -769,9 +772,11 @@ export class TimePickerFoundation implements ITimePickerFoundation {
     if (this._allowSeconds !== value) {
       this._allowSeconds = !!value;
       this._applyAllowSeconds();
-      if (this._isInitialized) {
+      if (this._isInitialized && !this._showMaskFormat) {
```

Comment:
this seems backwards, if not showMaskFormat then apply the mask?

Reply:
After re-evaluating. I still think this is correct. If this condition was not here, then applyMask function will make the mask show if showMaskFormat is true. When testing in the UI, turning on and off allowSeconds and use24HourTime will cause the mask to show when it should not. Still open to discussion on this one @madhater and @DRiFTy17